### DISCUSSION TITLE
Issue with keycloak password reset email delivery

### DISCUSSION THREAD

**Discussion by Lucas-PG** [keycloak/keycloak#32627] [Q&A]
I'm currently managing user passwords in Keycloak, where the users' emails are linked directly to their accounts. This means that when a user forgets their password and clicks on the "Forgot Password" button, the reset email is sent to the same email address associated with their account.

However, this creates a problem: if the user has forgotten their password, they cannot access the linked email to receive the reset instructions, effectively locking them out.

Given that changing the linked emails is not an option due to work-related constraints, what alternative solutions do I have to handle this situation? Any suggestions or best practices would be greatly appreciated.

I've searched for similar issues in various forums, but all I found were unanswered questions, [like this one](https://keycloak.discourse.group/t/sending-reset-password-notification-to-additional-mail-inbox/9829).

**Comment by RichardNeill** [comment]
We have the same issue: the primary email IS protected by Keycloak login.
What we'd like to do is have the password reset token sent to ALL the email addresses that we have for the user.

**Comment by Rohmilchkaese** [comment]
Push for that one! Could we please have someone look at this?

**Comment by Luktan82** [comment]
I tried to make it possible with an SPI Event Listener, but I struggled with the generated token in the second mail.
So I made it this way:
- User attribute in Keycloak for secondary mail address 
- postfix installed on another server
- declare this postfix server in Keycloak for Email SMTP
- in postfix after getting an email with the subject "reset password" a script is called against API of Keycloak to get the secondary mail address of the user
- Modified the mail to input the secondary mail address in CC field
- Send the modified mail to relay server (Exchange)

Everything works like a charme. :-)

If you want more Infos dont hesitate to get in touch with me.

Best regards
